frontend/public/components/edit-yaml.jsx

```diff
@@ -50,13 +50,26 @@ const stateToProps = ({ k8s, UI }) => ({
   models: k8s.getIn(['RESOURCES', 'models']),
 });
 
+const WithYamlTemplates = (Component) =>
+  function Comp(props) {
+    const [templateExtensions, resolvedTemplates] = useResolvedExtensions(
+      (e) => isYAMLTemplate(e) && e?.properties?.model?.kind === props?.obj?.kind,
```

Comment:
Do you need the `?` optional checks?
Definitely the extension itself must conform and have the model and kind.

```suggestion
      (e) => isYAMLTemplate(e) && e.properties.model.kind === props.obj.kind,
```

Looks like this only checks the `kind` and not `group` and `version`. Why not?

Reply:
`group` and `version` are not consistent. 
for PVC for example the `apiVersion` is `apiVersion: v1`
for VMTemplate its `kubevirt.io/v1alpha3`

some use only version, some use group/version
why kind is not enough here? AFAIK we don't have multiple templates for the same kind

Reply:
Oh that's interesting. We should track this as a separate issue then because that seems like a bug somewhere.
At the end of the day, this filtering at the extension level is only the first step. Even if we resolve all extensions, it still works. So for now just checking kind will work and will already reduce the number of code refs we need to resolve.

Reply:
I haven't reviewed the PR closely so I don't know the context, but `kind` is not unique on its own. We should be checking the group everywhere as well. The core k8s types that existed before there were API groups only have the version in `apiVersion` without a group (because it is not in a group). New types will all have groups, however.